### PR TITLE
Add support for Multipart with JSON parts

### PR DESCRIPTION
**Pull request by bluk**
- Add support for Multipart with JSON parts

### CODE REVIEW

File: lib/BraintreeHttp/Serializer/Multipart.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+        return implode("\r\n", $body);
+    }
+
+    public function decode($data)
+    {
+        throw new \Exception("Multipart does not support deserialization");
+    }
+
+    private function isAssociative(array $array) {
```
Comment (quinnjn) [inline lib/BraintreeHttp/Serializer/Multipart.php]: Bracket styling mismatch in this file

File: lib/BraintreeHttp/Serializer/Multipart.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    private function prepareFormField($partName, $value, $boundary) {
+        return implode("\r\n", [
+            "Content-Disposition: form-data; name=\"{$partName}\"",
+            "",
+            filter_var($value),
+        ]);
+    }
+
+    private function prepareFilePart($partName, $file, $boundary) {
+        // $file = fopen(dirname(__DIR__), '/file/path.png', 'rb')
```
Comment (quinnjn) [inline lib/BraintreeHttp/Serializer/Multipart.php]: Commented out code

**Review by quinnjn — COMMENTED** [review]
Couple styling notes but otherwise its 👍

File: lib/BraintreeHttp/Serializer/Multipart.php [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        // add boundary for each parameters
+        array_walk($body, function (&$part) use ($boundary) {
+            $part = "--{$boundary}\r\n{$part}";
+        });
+
+        // add final boundary
+        $body[] = "--{$boundary}--";
+        $body[] = "";
+
+        return implode("\r\n", $body);
```
Comment (demerino) [inline lib/BraintreeHttp/Serializer/Multipart.php]: Maybe make a `LINE_FEED` constant for `\r\n`?

**Review by demerino — APPROVED** [review]
Looks good, just small syntax things that were mentioned.

**Comment by demerino** [comment]
👍

**Review by quinnjn — APPROVED** [review]

**Comment by bluk** [comment]
One last change to ensure that the request headers are set correctly if they were modified by the serializer. In this case to add the `; boundary=--...`

**Pull request merged**